packages/extension-api/src/extension-api.d.ts

```diff
@@ -2488,4 +2488,24 @@ declare module '@podman-desktop/api' {
      */
     export function createCliTool(options: CliToolOptions): CliTool;
   }
+
+  export interface ImageCheckPartialResult {
+    category: string;
+    description: string;
+  }
+
+  export interface ImageCheckResult {
+    status: string;
+    partialResults: ImageCheckPartialResult[];
+  }
+
+  export interface ImageCheckerProvider {
+    readonly name: string;
+    readonly categories: string[];
+    checkImage(image: string): Promise<ImageCheckResult>;
```

Comment:
AFAIK the provider goal is to provide only functions so I don't see name and categories there ?

Reply:
I think checkImage is missing a `cancellationToken: CancellationToken` parameter to cancel the check

Reply:
The `name` is what the UI can display for this provider, for example "Image OpenShift Analyzer". The `categories` indicates to the UI which are the different categories of tests the checker will run, so the results can be classified in these categories (for example `USER directive`, `RUN Directive`, `Expose directive`)

Reply:
it looks that from design perspective, the static fields are passed on the registration method

example of contract:

```
registerCodeActionsProvider(selector: DocumentSelector, provider: CodeActionProvider, metadata?: CodeActionProviderMetadata): Disposable;
```

From end user perspective, I don't really understand at all the goals of `USER`, `RUN` or `Expose directive`

I mean, I launch a check on the image so if the provider is there it'll do the check, else it won't.

then, if the checker is not able to analyze some images it's more a selector/filter parameter

About the name, it's part of the Result no ?

Reply:
![design-checker](https://github.com/containers/podman-desktop/assets/9973512/b74875ce-6478-4f2b-8450-3427660e8f32)

In this example, the name would be `OpenShift Checker`, the UI needs to know it before to start the tests, to be able to display it in the tab.

For the categories, they would be `Container User` and `Exposed Ports`, so the UI can display them without any failed / passed mark, the time the checks are executed, then they are updated with passed / failed. But I perhaps misunderstood the design doc.

Reply:
I read again draft of the spec

For categories, something like 'Container User' or 'Exposed Ports' are only known by the checker and to me it's part of the result of the scan, not part of the registration.

Until we didn't launch a scan, nothing is displayed except that we're processing.

> In this example, the name would be OpenShift Checker, the UI needs to know it before to start the tests, to be able to display it in the tab.

what we can do for that is to give an extra argument when registering the provider with metadata

```ts
registerImageCheckerProvider(imageCheckerProvider: ImageCheckerProvider, metadata?: ImageCheckerProviderMetadata): Disposable;
```

and in the metadata you can specify a readonly name

without the name, we can assume the provider will take the name of the extension

so if your extension is named 'OpenShift Checker' then it'll use that name as well



Now if I look at the signature of the provider, it should return a ProviderResult<ImageCheck> result (or ImageCheck as the type we want is ImageCheck, result is just from what we call)

then if I look at Trivy extension result
![image](https://github.com/containers/podman-desktop/assets/436777/b2b26182-5cce-4939-9d68-45e2ce40a74d)

I got them classified/sorted by severity


but if I look at

```ts
  export interface ImageCheckPartialResult {
    category: string;
    description: string;
  }

  export interface ImageCheckResult {
    status: string;
    partialResults: ImageCheckPartialResult[];
  }
```

>  (a partial result indicates a failing check on the image).

I don't get the why a partial result would be a failed result. I assume partial result is that I don't have all the result, not that it failed

so looking back at trivy results and openshift checker, it looks like I need a severity (like using a wrong EXPOSE stuff might be less critical than using an image with root user)


I would probably need some markdown result like if it passes the test

![image](https://github.com/containers/podman-desktop/assets/436777/0ce7f7c1-1c4a-4d32-a34d-c701a46d2e5b)


and then probably a score or grade if it fails.

my image is Grade A, grade B, grade C etc depending on the check

Reply:
thanks @benoitf for your feedback. I'll make changes based on this